tests/modules/systemd/services.nix

```diff
@@ -23,6 +26,9 @@ with lib;
             Environment=VAR1=1
             Environment=VAR2=2
             ExecStart=/some/exec/start/command --with-arguments "%i"
+            ExecStartPre=${
+              builtins.unsafeDiscardStringContext drvScript.outPath
```

Comment:
Is `unsafeDiscardStringContext` necessary here?

I'm not well versed in the dark arts of those low-level operations, if you have an explanation for why you need it I'd gladly hear it.

Reply:
Could probably use `pkgs.writeText` instead of `builtins.toFile`, that way you wouldn't need it.

Reply:
The script is written using `pkgs.writeShellScript` which used `pkgs.writeTextFile` (just like `pkgs.writeText`) unless I am mistaken.

Reply:
Good point, it's actually nicer to use `pkgs.writeText` indeed, I made the change now.

@ambroisie `builtins.toFile` was used to create the file with the expected output, so that's the one who was complaining about a store path (e.g. a string with context, in nix parlance) being included in its output, nix forbids this in general.